cmd/controller/controller.go

```diff
@@ -85,7 +85,17 @@ func (c *ZVController) syncZV(zv *apis.ZFSVolume) error {
 			zvol.RemoveZvolFinalizer(zv)
 		}
 	} else {
-		err = zvol.SetZvolProp(zv)
+		// if finalizer is not set then it means we are creating
```

Comment:
Will there be a case where finalizer is removed as part of deletion. should also check for deletion flag to ignore the event?

Reply:
we are already doing that in the "if" check. this is else of that.